### DISCUSSION TITLE
How to get workspaceObjectId when user opens an existing item

### DISCUSSION THREAD

**Discussion by dhrumil83** [microsoft/Microsoft-Fabric-workload-development-sample#110] [Q&A] [answered]
[Route.targetUrl ](https://github.com/microsoft/Microsoft-Fabric-workload-development-sample/blob/main/Frontend/src/index.ui.tsx#L15) gives me the workspaceObjectId during the new item workflow.

But when I open an existing item, I get the itemId. For the workflow where the user wants to save a copy or go to a new empty item, seems like I need access to the workspaceObjectId. 
I could not quite find it in the workloadClient, so wondering if it's available somewhere.

**Comment by ramizsha** [comment, marked as answer]
Hi @dhrumil83,
If you want to know for a specific item, in which workspace it is located, you can use the [getItem API](https://learn.microsoft.com/en-us/javascript/api/@ms-fabric/workload-client/itemcrudapi?view=fabric-extensibility-js-latest#@ms-fabric-workload-client-itemcrudapi-getitem).
The result will contain the folderObjectId of the item which is the workspaceObjectId.

**Comment by dhrumil83** [comment]
thanks @ramizsha that works.

1 more related question: User opens a new item.
e.g. `workloadClient.navigation.navigate("workload", { path:  "AppName/${workspaceObjectId}"});`

Makes changes but wants to reset the view. When I call the same path via navigate, it does not call `workloadClient.navigation.onNavigate`

Can I force this navigate call, so it's like a browser refresh.
It's easier to reset everything at the top, rather than keeping track of everything in the app.

Note: I can force the refresh by adding something extra(timestamp) to the path, but wondering if there is a recommended approach.
e.g. workloadClient.navigation.navigate("workload", { path:  "AppName/${workspaceObjectId}/${timestamp}"});

**Comment by michal-mano** [comment]
Hi, we don't have a specific recommendation for refreshing the page. However, I think using the navigation API for cleanup is probably not best practice, and I would implement a cleanup function instead that resets the view.
